src/compressed_tensors/quantization/utils/helpers.py

```diff
@@ -92,22 +84,23 @@ def calculate_qparams(
     bit_min, bit_max = calculate_range(quantization_args, device)
     bit_range = bit_max - bit_min
 
-    if is_fp4(quantization_args=quantization_args):
-        zp_dtype = FP8_E4M3_DATA.dtype
-    else:
-        zp_dtype = quantization_args.pytorch_dtype()
+    zp_dtype = quantization_args.zp_dtype
 
     if quantization_args.symmetric:
         max_val_pos = torch.max(torch.abs(min_vals), torch.abs(max_vals))
+        scales = max_val_pos / (float(bit_range) / 2)
 
-        if is_fp4(quantization_args=quantization_args) and global_scale is not None:
+        if global_scale is not None:
             # Conditionally scale the generated local scale by a global_scale
-            scales = global_scale * (max_val_pos / FP4_E2M1_DATA.max)
-            scales = torch.clamp(scales, max=FP8_E4M3_DATA.max, min=FP8_E4M3_DATA.min)
-            scales = scales.to(FP8_E4M3_DATA.dtype)
+            scales = global_scale * scales
 
-        else:
-            scales = max_val_pos / (float(bit_range) / 2)
+        if quantization_args.scale_dtype is not None:
+            scales = torch.clamp(
+                scales,
+                max=torch.finfo(quantization_args.scale_dtype).max,
```

Comment:
will fail if scale_dtype is not a float. Consider making/finding a util to get max/min agnostic of float/int